pulseapi/entries/views.py

```diff
@@ -4,13 +4,48 @@
 
 import django_filters
 from rest_framework import (filters, status)
-from rest_framework.decorators import detail_route
-from rest_framework.generics import ListCreateAPIView, RetrieveAPIView
+from rest_framework.decorators import detail_route, api_view
+from rest_framework.generics import ListCreateAPIView, RetrieveAPIView, ListAPIView
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 
 from pulseapi.entries.models import Entry
 from pulseapi.entries.serializers import EntrySerializer
+from pulseapi.users.models import UserBookmarks
+
+@api_view(['PUT'])
+def toggle_bookmark(request, entryid):
```

Comment:
After thinking about it some more, entries already say whether they are bookmarked or not when you call `/entries` so for regular operation we don't need a "toggle yes or no", the only time we *would* need that is for bulk bookmark syncing, which we should probably do on its own dedicated route so a full set of bookmarks can be sent over in a single call.